### ISSUE TITLE
ComposedModal does not have programmatically associated name

### ISSUE THREAD

**Issue by masterDev1985** [carbon-design-system/carbon#6971]
I got hit with [an AVT issue](https://github.ibm.com/IBM-Blockchain/OpTools/issues/4376) on the ComposedModal I was using.  I guess when `role=dialog` was set on the inner `div` in https://github.com/carbon-design-system/carbon/issues/6122, an `aria` or `title` attribute should also have been set.

##  Environment

> Operating system

OpenClient
```
Linux oc8878866624.ibm.com 3.10.0-1127.19.1.el7.x86_64 #1 SMP Tue Aug 11 19:12:04 EDT 2020 x86_64 x86_64 x86_64 GNU/Linux
```

> Browser

Chrome 85.0.4183.83

> Automated testing tool and ruleset

N/A

> Assistive technology used to verify

IBM Equal Access Accessibility Checker

## Detailed description

> What version of the Carbon Design System are you using?

`carbon-components-react@7.20.0`

> What did you expect to happen?

The ComposedModal should follow these accessibility guidelines:
- Use the aria-labelledby attribute to reference existing visible text on the page that can clearly identify the purpose of the component;
- OR use the aria-label attribute with a short description as its value that can clearly dentify the purpose of the component when a visual label is not available;
- OR use the title attribute to provide a name when the design cannot accommodate a visual label;
- OR if the element's role allows it to derive a label from its content, provide the label text as the inner content of the element (e.g. <button type="submit">Search</button>).

> What happened instead?

See title.

> What WCAG 2.1 checkpoint does the issue violate?

[Checkpoint 4.1.2 Name, Role, Value](https://www.ibm.com/able/guidelines/ci162/name_role_value.html)

## Steps to reproduce the issue

1. Go to https://react.carbondesignsystem.com/?path=/story/composedmodal--using-child-nodes
2. Open the IBM Equal Accessibility Checker and click Scan.

## Additional information
![image](https://user-images.githubusercontent.com/14877096/95090249-0c699e00-06f3-11eb-989d-4562195a5c95.png)

![image](https://user-images.githubusercontent.com/14877096/95089884-aa109d80-06f2-11eb-8aa5-51d5d43fb2f3.png)
![image](https://user-images.githubusercontent.com/14877096/95089962-c01e5e00-06f2-11eb-9fc9-d1584835add4.png)

**Comment by Manav802** [comment]
I have spent some time and read through the resources I understand what accessibility enhancement you wish to achieve by adding the aria label can I work on this, this would be my first contribution to Hactoberfest and carbon both.

**Comment by tw15egan** [comment]
Sure thing @Manav802, I'll assign you

**Comment by jsehull** [comment]
Hey team - I've been looking over this for a while and will be sharing some findings soon

**Comment by jsehull** [comment]
I hope I'm not missing anything 😅, but I believe this issue has been resolved already due to the natural progression of Carbon since the work in Issue [#6112](https://github.com/carbon-design-system/carbon/issues/6122)

@tw15egan No repository changes have occurred. Please let me know if this needs to be submitted in a different format.
_____
Testing findings:

1. No longer able to reproduce Accessibility Checker error
> Interactive component does not have a programmatically associated name
![image](https://user-images.githubusercontent.com/9935383/219106214-9665dbb0-86ce-4eca-84b9-ac6f4ec581a8.png)

2. Current code already has the requested updates
- view `packages/react/src/components/ComposedModal/ComposedModal.js`
- specifically has BOTH `aria-label` and `aria-labelledby`
![image](https://user-images.githubusercontent.com/9935383/219106148-81e33ac4-20c6-4801-b058-f831314f31b0.png)

3. Carbon versions have updated
- Then: `carbon-components-react@7.20.0`
- Now: `@carbon/react@1.22.0`
![image](https://user-images.githubusercontent.com/9935383/219106335-4cce5ce9-a745-4db4-8b52-02f5b5e7b254.png)


It seems to have been resolved along the way in the last 2.5 years. This is an odd one for me, please let me know if I'm missing something!

**Comment by tw15egan** [comment]
I'm also not seeing any a11y violations. Thanks for the deep dive and verifying, @jsehull! We can go ahead and close this one out
